extensions/binding/org.eclipse.smarthome.binding.coapconnectedlighting/src/main/java/org/eclipse/smarthome/binding/coapconnectedlighting/handler/CoAPConnectedLightingHandler.java

```diff
@@ -0,0 +1,361 @@
+
+        updateState(stringChannelTemperatureUID, new DecimalType(temperature));
+
+        logger.debug(clientResponseTemperature.advanced().getSource().toString());
+    }
+
+    private void getHumidity() {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/sensors/humidity");
+
+        logger.debug(coapClient.getURI());
+        CoapResponse clientResponseHumidity = coapClient.get(TEXT_PLAIN);
+        String humidityText = clientResponseHumidity.getResponseText();
+        String humidityFirstHalf = humidityText.substring(0, humidityText.lastIndexOf(','));
+        String humidity = humidityFirstHalf.substring(humidityFirstHalf.lastIndexOf(',') + 1);
+
+        updateState(stringChannelHumidityUID, new DecimalType(humidity));
+
+        logger.debug(clientResponseHumidity.advanced().getSource().toString());
+    }
+
+    private void getPressure() {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/sensors/pressure");
+
+        logger.debug(coapClient.getURI());
+        CoapResponse clientResponsePressure = coapClient.get(TEXT_PLAIN);
+        String pressureText = clientResponsePressure.getResponseText();
+        String pressureFirstHalf = pressureText.substring(0, pressureText.lastIndexOf(','));
+        String pressure = pressureFirstHalf.substring(pressureFirstHalf.lastIndexOf(',') + 1);
+
+        updateState(stringChannelPressureUID, new DecimalType(pressure));
+
+        logger.debug(clientResponsePressure.advanced().getSource().toString());
+    }
+
+    private void putLCD(String message) {
+        coapClient.setURI("coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/COAP/LCDResource");
+        CoapResponse clientResponseLCD = coapClient.put(message, TEXT_PLAIN);
+
+        logger.debug(clientResponseLCD.advanced().getSource().toString());
+    }
+
+    private void putLightOnOff(String message) {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/actuators/actuator1");
+        StringBuilder coapLightSwitch = new StringBuilder();
+        if (message.compareTo("ON") == 0) {
+            coapLightSwitch.append(",1,pp,100,");
+        } else if (message.compareTo("OFF") == 0) {
+            coapLightSwitch.append(",1,pp,0,");
+        }
+        CoapResponse clientResponse = coapClient.put(coapLightSwitch.toString(), TEXT_PLAIN);
+
+        logger.debug(clientResponse.advanced().getSource().toString());
+    }
+
+    private void putlightDimmer(String message) {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/actuators/actuator1");
+        StringBuilder coapLightSwitch = new StringBuilder();
+        coapLightSwitch.append(",1,pp," + message + ",");
```

Comment:
having a StringBuilder and not using its #append method feels a little bit strange. You may use
```
coapLightSwitch.append(",1,pp,").append(message).append(",");
```

Reply:
@htreu I should get a beating for this one !